src/datatrove/pipeline/dedup/minhash.py

```diff
@@ -152,6 +152,8 @@ def read_sigs(
         index_file: is index file
     """
     with file as f:
+        if f.size == 0:
+            yield None
```

Comment:
```suggestion
            return
```

Reply:
I don't think we can simply return None here since the value of this function is used as an iterator in latter code

https://github.com/huggingface/datatrove/blob/a98aafd2f3fe3dab3addd8ad9483338e92494938/src/datatrove/pipeline/dedup/minhash.py#L386-L402

Reply:
It should still work, as long as you have a `yield` inside the function body it should be treated as a generator even if the yield is never reached

Reply:
Magic